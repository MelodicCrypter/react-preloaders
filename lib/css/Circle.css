body{overflow-x: hidden} #preloader{z-index:9999;position:fixed;top:0;left:0;height:100vh;width:100vw;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-webkit-align-items:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-transition:opacity .3s linear, visibility .2s linear .3s;transition:opacity .3s linear, visibility .2s linear .3s;opacity:1}#preloader.close{opacity:0;visibility:hidden}.circular{-webkit-animation:rotate 2s linear infinite;animation:rotate 2s linear infinite;height:50px;left:50%;position:absolute;top:50%;-webkit-transition:all 0.2s ease;transition:all 0.2s ease;-webkit-transform:translate(-50%, -50%) rotate(360deg);-ms-transform:translate(-50%, -50%) rotate(360deg);transform:translate(-50%, -50%) rotate(360deg);width:50px;z-index:4}.circular .path{stroke-dasharray:1, 500;stroke-dashoffset:0;-webkit-animation:dash 1.5s ease-in-out infinite, color 6s ease-in-out infinite;animation:dash 1.5s ease-in-out infinite, color 6s ease-in-out infinite;stroke-linecap:round}@-webkit-keyframes rotate{0%{-webkit-transform:translate(-50%, -50%) rotate(0deg);transform:translate(-50%, -50%) rotate(0deg)}100%{-webkit-transform:translate(-50%, -50%) rotate(360deg);transform:translate(-50%, -50%) rotate(360deg)}}@keyframes rotate{0%{-webkit-transform:translate(-50%, -50%) rotate(0deg);transform:translate(-50%, -50%) rotate(0deg)}100%{-webkit-transform:translate(-50%, -50%) rotate(360deg);transform:translate(-50%, -50%) rotate(360deg)}}@-webkit-keyframes dash{0%{stroke-dasharray:1, 200;stroke-dashoffset:0}50%{stroke-dasharray:89, 200;stroke-dashoffset:-35}100%{stroke-dasharray:89, 200;stroke-dashoffset:-124}}@keyframes dash{0%{stroke-dasharray:1, 200;stroke-dashoffset:0}50%{stroke-dasharray:89, 200;stroke-dashoffset:-35}100%{stroke-dasharray:89, 200;stroke-dashoffset:-124}}
