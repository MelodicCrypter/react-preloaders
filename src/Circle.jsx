import React, { Component } from 'react';
// import './css/Circle.css'; 

class Circle extends Component {
  componentDidMount() {
    document.onreadystatechange = () => {
      if (document.readyState === 'complete') {
        setTimeout(() => {
          document.getElementById('preloader').classList.add('close');
        }, this.props.time);
      }
    };
  }

  render() {
    return (
      <div style={{ backgroundColor: this.props.bgColor }} id="preloader">
        <svg className="circular" height="50" width="50">
          <circle
            className="path"
            cx="25"
            cy="25"
            r="20"
            stroke={this.props.color}
            fill="none"
            strokeWidth="2.5"
            strokeMiterlimit="10"
          />
        </svg>
      </div>
    );
  }
}

Circle.defaultProps = { time: 1300, bgColor: '#f7f7f7', color: '#2D2D2D' };

export default Circle;
