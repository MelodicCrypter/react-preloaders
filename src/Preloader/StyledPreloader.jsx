import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PreloaderStyles from './PreloaderStyles';

const StyledPreloader = styled.div`
  ${PreloaderStyles}
`;

function Preloader({
  children, background, color, time, customLoading, animation,
}) {
  const [loading, setLoading] = useState(true);
  const childrenWithProp = React.Children.map(children, child => React.cloneElement(child, {
    color,
  }));
  const bodyScroll = () => {
    document.body.style.overflow = loading ? 'hidden' : null;
    document.body.style.height = loading ? '100%' : null;
    document.body.style.width = loading ? '100%' : null;
    document.body.style.position = loading ? 'fixed' : null;
  };
  const generateAnimation = () => {
    const animate = animation && animation.split('-');
    return {
      name: animate && animate[0],
      direction: animate && animate[1],
    };
  };

  bodyScroll();
  useEffect(() => {
    if (customLoading === false) {
      setTimeout(() => {
        setLoading(false);
      }, time);
    }
    if (customLoading === undefined) {
      document.onreadystatechange = () => {
        if (document.readyState === 'complete') {
          setTimeout(() => {
            setLoading(false);
          }, time);
        }
      };
    }
  }, [customLoading]);
  return (
    <StyledPreloader
      animation={generateAnimation()}
      background={background}
      loadingStatus={loading}
      id="preloader"
    >
      {childrenWithProp}
    </StyledPreloader>
  );
}

Preloader.propTypes = {
  time: PropTypes.number,
  background: PropTypes.string,
  color: PropTypes.string,
  animation: PropTypes.string,
  children: PropTypes.element,
  customLoading: PropTypes.bool,
};

Preloader.defaultProps = {
  time: 1300,
  background: '#f7f7f7',
  color: '#2D2D2D',
  animation: 'fade',
};

export default Preloader;
