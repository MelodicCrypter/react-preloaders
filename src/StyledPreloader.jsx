import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

function StyledPreloader({ children, bgColor, color }) {
  const [loading, setLoading] = useState(true);
  const Preloader = styled.div`
    background: ${bgColor};
    z-index: 9999;
    position: fixed;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
    justify-content: center;
    align-items: center;
    display: flex;
    transition: opacity 0.3s linear, visibility 0.2s linear 0.3s;
    opacity: 1;
    visibility: 1;
  `;
  const bodyScroll = () => {
    document.body.style.overflow = loading ? 'hidden' : 'auto';
    document.body.style.height = loading ? '100%' : 'auto';
    document.body.style.width = loading ? '100%' : 'auto';
    document.body.style.position = loading ? 'fixed' : 'unset';
  };
  bodyScroll();
  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, []);
  useEffect(() => {});
  const childrenWithProp = React.Children.map(children, child => React.cloneElement(child, {
    color,
  }));
  return <Preloader id="preloader">{childrenWithProp}</Preloader>;
}

StyledPreloader.defaultProps = { time: 1300, bgColor: '#f7f7f7', color: '#2D2D2D' };

export default StyledPreloader;
