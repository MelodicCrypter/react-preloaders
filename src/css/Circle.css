body {
  overflow-x: hidden;
}
#preloader {
  z-index: 9999;
  position: fixed;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  justify-content: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  transition: opacity 0.3s linear, visibility 0.2s linear 0.3s;
  opacity: 1;
}
#preloader.close {
  opacity: 0;
  visibility: hidden;
}
.circular {
  animation: rotate 2s linear infinite;
  height: 50px;
  left: 50%;
  position: absolute;
  top: 50%;
  transition: all 0.2s ease;
  transform: translate(-50%, -50%) rotate(360deg);
  width: 50px;
  z-index: 4;
}
.circular .path {
  stroke-dasharray: 1, 500;
  stroke-dashoffset: 0;
  animation: dash 1.5s ease-in-out infinite, color 6s ease-in-out infinite;
  stroke-linecap: round;
}
@-webkit-keyframes rotate {
  0% {
    -webkit-transform: translate(-50%, -50%) rotate(0deg);
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100% {
    -webkit-transform: translate(-50%, -50%) rotate(360deg);
    transform: translate(-50%, -50%) rotate(360deg);
  }
}
@keyframes rotate {
  0% {
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100% {
    transform: translate(-50%, -50%) rotate(360deg);
  }
}
@-webkit-keyframes dash {
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -124;
  }
}
@keyframes dash {
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -124;
  }
}
