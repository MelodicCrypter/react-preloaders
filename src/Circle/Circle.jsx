import React from 'react';
import StyledPreloader from '../StyledPreloader';

const Circle = ({ className, color }) => (
  <svg className={className} height="50" width="50">
    <circle
      className="path"
      cx="25"
      cy="25"
      r="20"
      stroke={color}
      fill="none"
      strokeWidth="2.5"
      strokeMiterlimit="10"
    />
  </svg>
);
const CircleWithPreloader = ({ className }) => (
  <StyledPreloader>
    <Circle className={className} />
  </StyledPreloader>
);

export default CircleWithPreloader;
