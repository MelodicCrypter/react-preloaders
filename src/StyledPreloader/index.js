import StyledPreloader from './StyledPreloader';

export default StyledPreloader;
