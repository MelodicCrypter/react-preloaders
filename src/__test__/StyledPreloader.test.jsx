import React from 'react';
import { shallow } from 'enzyme';
import StyledPreloader from '../StyledPreloader';
import Circle from '../Circle/StyledCircle';
import 'jest-styled-components';

describe('StyledPreloader', () => {
  it('renders correctly', () => {
    const component = shallow(
      <StyledPreloader>
        <Circle />
      </StyledPreloader>,
    );
    expect(component).toMatchSnapshot();
  });
});
