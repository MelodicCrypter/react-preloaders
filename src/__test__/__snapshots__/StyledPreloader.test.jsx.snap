// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledPreloader renders correctly 1`] = `
<styled.div
  bgColor="#f7f7f7"
  id="preloader"
  loadingStatus={true}
>
  <Styled(CircleWithPreloader)
    color="#2D2D2D"
    key=".0"
  />
</styled.div>
`;
